GObject-2.0.Integration fails for classes in global namespace

If a class annotated with "[Subclass<GObject.Object>]" is in the global namespace, then the GObject-2.0.Integration source generator fails.  The generated .cs source file fails to compile, with an error like this:

```
GObject-2.0.Integration/GObject.Integration.SourceGenerator.Generator/ItemData.Subclass.g.cs(1,11): error CS1001: Identifier expected
```

That's because the generated .cs file begins with this, which is not valid C#:

```
namespace <global namespace>;
```

To see this problem, edit the GridView sample by removing the line "namespace GridViewSample;" from CustomObjectGridViewWindow.cs and StringListGridViewWindow.cs, then try to build it.

--- a/src/Extensions/GObject-2.0.Integration/SourceGenerator/Subclass/SubclassCode.cs
+++ b/src/Extensions/GObject-2.0.Integration/SourceGenerator/Subclass/SubclassCode.cs
@@ -15,11 +15,12 @@ internal static class SubclassCode
     }
     private static string ToCode(SubclassData subclassData)
     {
-        return $$"""
-               namespace {{subclassData.Namespace}};
-               
-               {{RenderClassHierarchy(subclassData)}}
-               """;
+        return subclassData.IsGlobalNamespace
+            ? RenderClassHierarchy(subclassData)
+            : $"""
+                namespace {subclassData.Namespace};
+                {RenderClassHierarchy(subclassData)}
+                """;
     }
 
     private static string RenderClassHierarchy(SubclassData subclassData)

--- a/src/Extensions/GObject-2.0.Integration/SourceGenerator/Subclass/SubclassData.cs
+++ b/src/Extensions/GObject-2.0.Integration/SourceGenerator/Subclass/SubclassData.cs
@@ -8,6 +8,7 @@ internal sealed record SubclassData(
     string Parent,
     string ParentHandle,
     string Namespace,
+    bool IsGlobalNamespace,
     string Accessibility,
     string FileName,
     Stack<TypeData> UpperNestedClasses

--- a/src/Extensions/GObject-2.0.Integration/SourceGenerator/Subclass/SubclassValuesProvider.cs
+++ b/src/Extensions/GObject-2.0.Integration/SourceGenerator/Subclass/SubclassValuesProvider.cs
@@ -49,6 +49,7 @@ internal static class SubclassValuesProvider
             Parent: parentType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat),
             ParentHandle: parentHandle,
             Namespace: context.TargetSymbol.ContainingNamespace.ToDisplayString(),
+            IsGlobalNamespace: context.TargetSymbol.ContainingNamespace.IsGlobalNamespace,
             Accessibility: accessibility,
             FileName: GetFileName(subclass),
             UpperNestedClasses: upperNestedClasses

--- a/src/Tests/Libs/GirTest-0.1.Tests/SubclassIntegrationTest.GlobalNamespace.cs
+++ b/src/Tests/Libs/GirTest-0.1.Tests/SubclassIntegrationTest.GlobalNamespace.cs
@@ -1,0 +1,2 @@
+﻿[GObject.Subclass<GObject.Object>]
+internal partial class SomeGlobalSubClass;

--- a/src/Tests/Libs/GirTest-0.1.Tests/SubclassIntegrationTest.cs
+++ b/src/Tests/Libs/GirTest-0.1.Tests/SubclassIntegrationTest.cs
@@ -46,26 +46,34 @@ public class SubclassIntegrationTest : Test
         var type4 = SomeGenericSubclass<SomeGenericSubclass<string>>.GetGType();
         var type5 = SomeSubSubClass.GetGType();
         var type6 = SomeContainingClass.SomeNestedGenericSubSubSubClass.GetGType();
+        var type7 = SomeGlobalSubClass.GetGType();
 
         type1.Should().NotBe(type2);
         type1.Should().NotBe(type3);
         type1.Should().NotBe(type4);
         type1.Should().NotBe(type5);
         type1.Should().NotBe(type6);
+        type1.Should().NotBe(type7);
 
         type2.Should().NotBe(type3);
         type2.Should().NotBe(type4);
         type2.Should().NotBe(type5);
         type2.Should().NotBe(type6);
+        type2.Should().NotBe(type7);
 
         type3.Should().NotBe(type4);
         type3.Should().NotBe(type5);
         type3.Should().NotBe(type6);
+        type3.Should().NotBe(type7);
 
         type4.Should().NotBe(type5);
         type4.Should().NotBe(type6);
+        type4.Should().NotBe(type7);
 
         type5.Should().NotBe(type6);
+        type5.Should().NotBe(type7);
+
+        type6.Should().NotBe(type7);
     }
 }
 

